legacy/firmware/ethereum.c

```
@@ -444,14 +444,16 @@ static bool layoutEthereumConfirmTx(
     if (!is_eip1559) {
       return layoutTransactionSignEVM(
           chain_name, params->chain_id, false, amount, to_str, signer, NULL,
-          NULL, params->data_initial_chunk_bytes, data_total,
-          _(I__ETH_MAXIMUM_FEE_COLON), gas_value, _(I__TOTAL_AMOUNT_COLON),
-          total_amount, NULL, NULL, NULL, NULL);
+          NULL, params->data_initial_chunk_bytes,
+          data_total > 1024 ? 1024 : data_total, _(I__ETH_MAXIMUM_FEE_COLON),
+          gas_value, _(I__TOTAL_AMOUNT_COLON), total_amount, NULL, NULL, NULL,
+          NULL);
     } else {
```

Clang (14.0.6)

[note] +1, nesting level increased to 2

(clang)